### ISSUE TITLE
分词方式加入padding后结果会成为乱码

### ISSUE THREAD

**Issue by songyingxin** [THUDM/ChatGLM-6B#283]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

正常情况下modeling_chatglm的chat函数下的分词为：input_ids = tokenizer([prompt], return_tensors="pt", padding=True)， 此时输出的结果是没有问题的。 

当我将输入改为：input_ids = tokenizer([prompt], return_tensors="pt", padding="max_length", max_length=100)， 引入padding会，会返回乱码，如下：
![image](https://user-images.githubusercontent.com/13884292/228834535-cfc93ef4-accd-47cb-b953-b46a8cf6428a.png)

我认为是attention mask 的默认构造是不是有问题，我还没来得及看。

### Expected Behavior

期望正常就算引入padding后也能够正常输出。

### Steps To Reproduce

无

### Environment

```markdown
- OS:
- Python:
- Transformers:
- PyTorch:
- CUDA Support (`python -c "import torch; print(torch.cuda.is_available())"`) :
```

### Anything else?

无

**Comment by JaheimLee** [comment]
盲猜一下是不是padding side问题。正好引出这个问题，我看commit记录中把padding side给改到right了。训练时候应该无所谓，但batch推理时候必须要left padding吧
